src/sota_tools/ostree_object.cc

```diff
@@ -241,6 +246,10 @@ void OSTreeObject::Upload(const TreehubServer &push_target, CURLM *curl_multi_ha
 
   curlEasySetoptWrapper(curl_handle_, CURLOPT_PRIVATE, this);  // Used by ostree_object_from_curl
 
+  assert(headers_ == nullptr);
```

Comment:
If `OSTreeObject::Upload` is called more than one for the same class instance then it won't work. I suppose, in our case it's called only once so it shouldn't be a problem, although, perhaps makes sense to set the headers in ctor ?